src/transaction.js

```diff
@@ -177,10 +183,7 @@ class Transaction extends EventEmitter {
         }
         if (status === 2) {
           if (isUndefined(value)) {
-            if (
-              get(res, 'context.sql', '').toUpperCase() === 'ROLLBACK' &&
-              this.doNotRejectOnRollback
-            ) {
+            if (this.doNotRejectOnRollback && /^ROLLBACK\b/i.test(sql)) {
```

Comment:
Should this be case-sensitive?

Reply:
This change is probably also needed for db dialect transaction implementation.

Reply:
I mean https://github.com/tgriesser/knex/blob/master/src/dialects/mysql/transaction.js, doesn't look like any other dialect has this logic.

Reply:
- Current implementation is case-insensitive, so that's why i added `/i` flag to make the regexp case-insensitive too. But actually it does not need to be case-insensitive, since only valid way to rollback or commit is though `trx.rollback()` or `trx.commit()`, only they set `status` argument, so only they `_resolve()` or `_reject()`, so only they release a connection properly, and they use upper case, so actually there is no need for `/i` flag, i just kept it to be similar to current implementation.

- Other dialects do usual rollback fine, since only postgres was failing the test. But i assume they do nested rollback badly (first issue). So yeah, other dialects should be updated to treat `rollbackTo()` queries the same way as `rollback()`, but it will require new tests for such cases and new PR. I just wanted to make current tests pass.

Reply:
Fair enough!